src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java

```diff
@@ -895,7 +895,11 @@ protected void move(Block b, BlockFace face, Block block) {
 
             Slimefun.runSync(() -> {
                 PlayerSkin skin = PlayerSkin.fromBase64(texture);
-                PlayerHead.setSkin(block, skin, true);
+                Material type = block.getType();
+                // Ensure that this Block is still a Player Head
+                if (type == Material.PLAYER_HEAD || type == Material.PLAYER_WALL_HEAD) {
+                    PlayerHead.setSkin(block, skin, true);
+                }
```

Comment:
This is literally the code at the very start of `PlayerHead.setSkin(...)` so this should not change anything and be redundant??

Reply:
Ohhh, nevermind it does the check and then throws the exception 🤦🏻‍♂️
So yeah, this checks out...